### PR TITLE
add scripts to build a dev kind cluster and necessary tools

### PR DESCRIPTION
**Pull request by longfangsong**
## what type of PR is this?

- [FEATURE] add scripts to build a local kind cluster (inside or outside of a docker container environment) to simplify developing environment building.

## Addtional documentation

These scripts are copied from `tidb-operator` with some change (eg. add kustomize and remove helm)

### CODE REVIEW

File: hack/run-in-container.sh [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+Environments:
+
+    CLEANUP             if passed, clean up local caches
+    DOCKER_LIB_VOLUME   the name of docker lib volume, defaults: benchmark-operator-docker-lib
+    DOCKER_GRAPH_VOLUME the name of docker graph volume, defaults: benchmark-operator-docker-graph
+    DOCKER_GO_VOLUME    the name of go cache volume, defaults: benchmark-operator-go
+    NAME                the name of container
+
+Examples:
```
Comment (Yisaer) [inline hack/run-in-container.sh]: What purpose is `run-in-container` script for?

Comment (longfangsong) [reply hack/run-in-container.sh]: Using an isolated container environment for development makes it easier to do cleanup jobs when the environment messed up. And some tools used in the developing environemt might have different version in the developer's operating system, which might cause problems and confusion.
After all, using this script is optional, if anyone thinks it is not necessary, he can simply ignore this.

File: .gitignore [hunk 2]
```diff
@@ -22,3 +22,4 @@ bin
 *.swp
 *.swo
 *~
+output
```
Comment (Yisaer) [inline .gitignore]: Add line in the end

File: hack/run-in-container.sh (hunk 1 above)
Comment (Yisaer) [reply hack/run-in-container.sh]: It's ok to me.

File: .gitignore (hunk 2 above)
Comment (longfangsong) [reply .gitignore]: > Add line in the end

Maybe I should add the script to fix this kind of problem, and it'll be more useful after the CI is ready.

**Pull request merged**